### PR TITLE
Adds Ethereal Drinks

### PR DESCRIPTION
**Pull request by KingkumaArt**
## About The Pull Request
Adds 3 new electric-themed drinks for ethereals. They count as food for etherials, (not much, one glass is about equivelant to a plasma burger), but humans can drink them as well, among other effects.
## Why It's Good For The Game
As an ethereal player, I do feel while our food is convenient, that we miss out on the RP other species get from going to the bar. This allows for etherials to order drinks from the bartender. All effects work on all races, with the hunger satiation as the exception. 
## Changelog
:cl:
add: Voltaic Yellow Wine - New "base" drink, found in booze-o-mat. No special effects besides acting as a weak ethereal food. Not very potent in terms of alcohol.

add: Telepole - New mixed drink themed after thunderstorms, gives the same shock-resist grey bull does. Made from 1 part Voltaic Wine, 1 Part Sake, and 2 parts Dark & Stormy.  Moderately potent.

add: Pod Tesla - New mixed drink, themed after the old removed tesla engine (the singulo gets a cocktail, it should too!) Grants a brave - bull phobia resist, and a stronger grey bull shock resist allowing you to (temporarily) resist tesla arcs from reactive armor and the SM's tesla coils. Gives a pleasant thought to whoever drank it, because this thing is a pain to make. Made from 5 parts admiralty, 5 parts telepole, and 3 parts brave bull. Highly potent.

qol: Unsure if this counts as QOL or balance, but the fact sol dry is in 3 different cocktails but the bartender has to buy 30u cans of it made me feel it deserved being added to the soda dispenser.

image: Added graphics for the above drinks, shown below, from left to right: Pod tesla, Voltaic Yellow wine, Telepole, Voltaic Wine (bottle)
image

/:cl:

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
hi!

**Comment by newfren** [comment]
sol dry in the soda dispenser is so helpful it's such a pain in the ass using cans

**Comment by Likteer** [comment]
The prerequisites are mothic drinks?

**Comment by KingkumaArt** [comment]
> sol dry in the soda dispenser is so helpful it's such a pain in the ass using cans

Main reason I added it was I was getting sick of buying cans during testing for the dark & stormy lmao

**Comment by KingkumaArt** [comment]
> The prerequisites are mothic drinks?

These aren't meant to be etherial drinks in the way moffic drinks are, I just picked those two as they fit the theming - dark&stormy being themed around thunderstorms, and admiralty being the closest I could think of to a nanotrasen-inspired cocktail.

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 1]
```diff
@@ -3678,5 +3678,102 @@
...
+	..()
+
+/datum/reagent/consumable/ethanol/telepole/expose_mob(mob/living/exposed_mob, methods=TOUCH, reac_volume) //can't be on life because of the way blood works.
+	. = ..()
+	if(!(methods & (INGEST|INJECT|PATCH)) || !iscarbon(exposed_mob))
+		return
+
+	var/mob/living/carbon/exposed_carbon = exposed_mob
+	var/obj/item/organ/internal/stomach/ethereal/stomach = exposed_carbon.get_organ_slot(ORGAN_SLOT_STOMACH)
+	if(istype(stomach))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:3737]: You should check the type prior to assuming the stomach is said type.

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 2]
```diff
@@ -3678,5 +3678,102 @@
...
+	..()
+
+/datum/reagent/consumable/ethanol/pod_tesla/expose_mob(mob/living/exposed_mob, methods=TOUCH, reac_volume) //can't be on life because of the way blood works.
+	. = ..()
+	if(!(methods & (INGEST|INJECT|PATCH)) || !iscarbon(exposed_mob))
+		return
+
+	var/mob/living/carbon/exposed_carbon = exposed_mob
+	var/obj/item/organ/internal/stomach/ethereal/stomach = exposed_carbon.get_organ_slot(ORGAN_SLOT_STOMACH)
+	if(istype(stomach))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:3776]: same here

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 3]
```diff
@@ -3678,5 +3678,102 @@
...
+
+/datum/reagent/consumable/ethanol/telepole/on_mob_metabolize(mob/living/affected_mob)
+	..()
+	ADD_TRAIT(affected_mob, TRAIT_SHOCKIMMUNE, type)
+
+/datum/reagent/consumable/ethanol/telepole/on_mob_end_metabolize(mob/living/affected_mob)
+	REMOVE_TRAIT(affected_mob, TRAIT_SHOCKIMMUNE, type)
+	..()
+
+/datum/reagent/consumable/ethanol/telepole/expose_mob(mob/living/exposed_mob, methods=TOUCH, reac_volume) //can't be on life because of the way blood works.
```
[the remark this answers is not in the document]
Comment (KingkumaArt) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:3730]: > im confused by this statement why do you want it on life?
According to the code for liquid electricity, the etherial food thing doesnt work on touch

**Review by Twaticus — APPROVED** [review]

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 4]
```diff
@@ -3678,5 +3678,102 @@
...
+	name = "Pod Tesla"
+	desc = "Ride the lightning! Recharges etherials, suppresses phobias, and grants strong temporary shock resistance."
+	icon = 'icons/obj/drinks/mixed_drinks.dmi'
+	icon_state = "pod_tesla"
+
+/datum/reagent/consumable/ethanol/pod_tesla/on_mob_metabolize(mob/living/affected_mob)
+	. = ..()
+	ADD_TRAIT(affected_mob, TRAIT_SHOCKIMMUNE, type)
+	ADD_TRAIT(affected_mob, TRAIT_TESLA_SHOCKIMMUNE, type)
+	ADD_TRAIT(affected_mob, TRAIT_FEARLESS, type)
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:3760]: use `add_traits()` instead

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 5]
```diff
@@ -3678,5 +3678,102 @@
...
+
+
+/datum/reagent/consumable/ethanol/pod_tesla/on_mob_end_metabolize(mob/living/affected_mob)
+	REMOVE_TRAIT(affected_mob, TRAIT_SHOCKIMMUNE, type)
+	REMOVE_TRAIT(affected_mob, TRAIT_FEARLESS, type)
+	REMOVE_TRAIT(affected_mob, TRAIT_TESLA_SHOCKIMMUNE, type)
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:3766]: use `remove_traits()` instead. you could also get away with having a list of these traits on the reagent as well to slim down on copypasta

**Review by san7890 — CHANGES_REQUESTED** [review]
two minor things and this looks good to me. mark "Ready for Review" when addressed.

**Review by KingkumaArt — COMMENTED** [review]
I hope I did the syntax right on the add_traits

**Pull request merged**